### PR TITLE
Feature/terraform aws

### PR DESCRIPTION
**Pull request by ryane**
supersedes https://github.com/CiscoCloud/microservices-infrastructure/pull/328

adds aws terraform support

### CODE REVIEW

File: plugins/inventory/terraform.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def inner(*args, **kwargs):
         if attrs.get('publicly_routable', False):
             groups.append('publicly_routable')
 
+        groups.append(attrs['consul_dc'])  # datacenter
```
Comment (BrianHicks) [inline plugins/inventory/terraform.py]: this and the groups below should be removed. They're not necessary in the new scheme.

File: plugins/inventory/terraform.py [hunk 2]
```diff
@@ -169,6 +183,38 @@ def openstack_host(resource, tfvars=None):
...
+@parses('aws_instance')
+@calculate_mi_vars
+def aws_host(resource, tfvars=None):
+    name = resource['primary']['attributes']['tags.Name']
+    raw_attrs = resource['primary']['attributes']
+
+    groups = []
+
+    tags = parse_dict(raw_attrs, 'tags')
+    attrs = {
```
Comment (BrianHicks) [inline plugins/inventory/terraform.py]: this should be fleshed out to translate *all* the information that Terraform provides. I can do it pretty quickly if you like, just send me your `.tfstate` file.

File: test/inventory/aws.state.snapshot [hunk 3]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (BrianHicks) [inline test/inventory/aws.state.snapshot]: this file is specific to a deployment and shouldn't be checked in

File: test/inventory/gce.state.snapshot [hunk 4]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (BrianHicks) [inline test/inventory/gce.state.snapshot]: same with this one

File: test/inventory/test_terraform.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import terraform
```
Comment (BrianHicks) [inline test/inventory/test_terraform.py]: I've got a branch open to make complete tests for this, would you remove the tests here from the PR?

File: test/runAll.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
```
Comment (BrianHicks) [inline test/runAll.sh]: re: tests, same here.

**Comment by BrianHicks** [comment]
I haven't tested this myself (I don't have access to an AWS account to test), but it looks good to me and runs for Ryan. It just needs documentation (see the `feature/terraform-docs` branch)

**Comment by siddharthist** [comment]
This provisions currectly on AWS.

**Pull request merged**